### ISSUE TITLE
Create autonomous testing system

### ISSUE THREAD

**Issue by b21kurar** [HGustavs/LenaSYS#13617]
Create a file that can call all tests in a folder containing all test cases for the services.

It should present the results of the tests and what service is being tested

E.g -
courseedservice tests:
testName1: passed
testName2: fail
etc ....

**Comment by HGustavs** [comment]
My ideas for how to show this.

We need a way to compare to previous runs, so we see if the passed tests  changed in some manner

If many tests are run, is there a graphical representation that may work so you would not need to do any scrolling?

**Comment by b21hamwe** [comment]
My idea is that we do a similiar presentation of the data as we did with the log.php file.
A dropdown menu that includes all the tests so that not all show at one time.
A colum/row table that the tester gets to set passed or failed to in the "table box". 

A run() function that all the tests that gets iniated when the "mother test" (meaning this file) gets runned.  
If you want only want to run one test, you can comment out one of the "includes" (include once....). 

Do we want test data to be saved to the log database? 
(loglena6db).

**Comment by c21alest** [comment]
Yes, this testing system is similar to log.php. There will be a bunch of testing files, and each one will look something like this:

````
include_once "../Shared/test.php";

$testData = array(
'value1' => 'test',
'value2' => 'test',
'service' => 'https://cms.webug.se/root/G2/students/c21alest/LenaSYS/DuggaSys/courseedservice.php',
'serviceData' => serialize(array( // Data that service needs to execute function
'opt' => 'NEW',
'username' => 'usr',
'password' => 'pass',
)),
'test1Debug' => false, // If true more information of test will be displayed
'test2Debug' => true,
'test3Debug' => true,
);

testHandler($testData, false); // 2nd argument true = prettyprint (HTML) false = raw JSON, no debug mode is avalible when using prettyprint
````

This autonomous testing system is then going to test all these test-case files and the result is to be presented in a readable form. 

More information can also be found here: #13532

**Comment by b21hamwe** [comment]
First "outcast" for the mother test file and how we can present the data.
Wishes, recommendations and tips are welcome from the leaders. 

<img width="250" alt="image" src="https://github.com/HGustavs/LenaSYS/assets/102579060/8e5bda94-7cc3-4422-8ad8-d66325d18c85">

**Comment by d21erisu** [comment]
Looks great!

**Comment by c21alest** [comment]
Sounds great!

**Comment by b21hamwe** [comment]
Updated version with checkmark and "X" icons. Created via UTF-8 Dingbats. 

<img width="289" alt="image" src="https://github.com/HGustavs/LenaSYS/assets/102579060/bd12198c-f29e-48de-99cf-17c93d6ad384">

Awaiting real tests to be created so that i can start bind the test to the mother test and start printing data

**Comment by b21hamwe** [comment]
Managed to get the JSON data and print X mark or checkmark depending on the result from the test.
<img width="500" alt="image" src="https://github.com/HGustavs/LenaSYS/assets/102579060/bb721fda-8dc1-48d4-b033-b4f05e9747c5">
